src/libraries/Microsoft.PowerFx.Interpreter/EvalContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    internal class EvalContext
+    {
+        private int _currentCallDepth = 0;
+
+        // We can set this up to read from powerfx config, but it isn't nessecary yet.
+        private readonly int _maxCallDepth;
```

Comment:
>_maxCallDepth [](http://example.com/codeflow?start=29&length=13)

If we just pass this as a stack parameter, then we don't need any locking or decrement calls.